### ISSUE TITLE
subtitle codes are being stripped out of srt files

### ISSUE THREAD

**Issue by sweaver2** [iPzard/mkvtoolnix-batch-tool#23]
i'm seeing an issue where srt files that are being merged into an mkv are losing their subtitle code.

files in the image below work as expected when using mkvtoolnix:
![image](https://user-images.githubusercontent.com/42624300/111699358-3fad6e80-87f5-11eb-971c-20fc7bd2530a.png)

but using the batch tool, they all turn into unknown.

i just grabbed your most recent update, wondering if that might have something to do with it.

**Comment by iPzard** [comment]
Thanks for reporting this. When you say their subtitle code, what is it exactly that you're referring to?

**Comment by sweaver2** [comment]
Like eng at the end of that srt file, representing english.

Previously, I could put mkvs and srt files into the app, like the
previously attached image, and it would correctly associate an the srt to
the language, in the metadata of the mkv.

About 3 days ago, I started hitting an issue where, no matter what I did, I
was hitting a batch process error. So I upgraded to the latest version,
from the Jan 29th release. And now the files are created successfully, but
the subtitles are all 01,02,etc with no language association.

**Comment by sweaver2** [comment]
i've tried uninstalling and reverting back to the previous version (2.0.3), but i'm still seeing the same issue.
i'm wondering if there's some cache or something that needs clearing.

**Comment by iPzard** [comment]
Strange, the language is determined automatically by parsing the subtitle text, so you don't need to include the language name in the file name with the batch tool. But I'm not sure why it's showing up as "unknown", it should show as the appropriate language. 

Would you mind sharing the subtitle files you're having the issue with and I can take a closer look to see what's going on there?

**Comment by sweaver2** [comment]
here's an example file (extension changed from srt to txt for upload)
[First Man (2018)_Track07 - Copy.txt](https://github.com/iPzard/mkvtoolnix-batch-tool/files/6168883/First.Man.2018._Track07.-.Copy.txt)

**Comment by iPzard** [comment]
Thanks for sharing, I was able to reproduce the issue with the file you provided. It's adding the subtitle file to the MKV, but the language isn't determined/included for some reason. 

I'll dig into this some more as time permits and hopefully have a patch released here shortly.

**Comment by sweaver2** [comment]
awesome man, thanks.
your tool has been invaluable while working through my movies / tv shows (setting up individual jobs in mkvtoolnix is quite painful)

i had about 65 files come out with majority unknown subtitles, wound up demuxing the srts, running this tool to get language codes on the files (https://github.com/mdcollins05/srt-lang-detect), using your batch converter to remove all the subs from the mkv, and then running all the mkv / srt combos individually through mkvtoolnix.

**Comment by iPzard** [comment]
Sorry about the wait on this, been extremely busy lately but haven't forgotten.

**Comment by iPzard** [comment]
Removing stale bot activity.. still planning to address this.

**Comment by iPzard** [comment]
I made some revisions that appear to have fixed the issue, feel free to re-open or create a new issue if you continue to experience problems. The new release can be found here: [v2.1.3](https://github.com/iPzard/mkvtoolnix-batch-tool/releases/tag/v2.1.3)

![done](https://user-images.githubusercontent.com/8584126/119238256-51add500-baf6-11eb-8b7f-778858547396.png)
